lib/mutant/cache.rb

```diff
@@ -1,14 +1,17 @@
 module Mutant
   # An AST cache
   class Cache
-    include Equalizer.new, Adamantium::Mutable
+    include Concord.new(:file), Adamantium::Mutable
 
     # Initialize object
     #
+    # @param [Class<File>]
+    #
     # @return [undefined]
     #
     # @api private
-    def initialize
+    def initialize(*)
+      super
       @cache = {}
```

Comment:
Also your `#parse` method isn't under review, but you could probably push the cache population into the constructor, eg:

```ruby
def initialize(*)
  super
  @cache = ThreadSafe::Cache.new
end

def parse(path)
  @cache.fetch_or_store(path) do |path|
    Parser::CurrentRuby.parse(file.read(path))
  end
end
```